SwanContents/swancontents/filemanager/projects_mixin.py

```diff
@@ -62,10 +62,10 @@ def _contains_swan_folder_name(self, path):
 
         return False
 
-    def _dir_model(self, path, content=True):
+    async def _dir_model(self, path, content=True):
         """ When returning the info of a folder, add the info of the project to which it belong to (if inside a Project) """
 
-        model = super()._dir_model(path, content)
+        model = await super()._dir_model(path, content)
```

Comment:
General comment for this and similar changes in this file: here we are calling the `_dir_model` method of `HasTraits`, which is the superclass of `ProjectsMixin`. Is `HasTraits._dir_model` a coroutine? And if so, how is it possible that we were calling it before as a normal method (not awaited)?

Reply:
> It is defined here: https://github.com/jupyter-server/jupyter_server/blob/c67a46befdcc972436150e9876b5d25d3c33ea56/jupyter_server/services/contents/filemanager.py#L750

What inheritance relationship has the class you are pointing to with your link (AsyncFileContentsManager) with HasTraits, which is the superclass you are invoking in your code here?

Reply:
The object that calls the method is of type SwanEosFileManager, so the super()._dir_model() will resolve to one of the superclasses, in this case AsyncFileContentsManager.
![image](https://github.com/user-attachments/assets/cb596d9c-dae4-4da4-a7fe-3f222857bfe9)